### ISSUE TITLE
pip dep is required but not declared

### ISSUE THREAD

**Issue by dcramer** [olzhasar/pytest-watcher#18]
```
➜  ~/s/peated (feat/rewrite-api-in-python) ✗ ptw
Traceback (most recent call last):
  File "/home/dcramer/src/peated/.venv/lib/python3.11/site-packages/pytest_watcher/config.py", line 9, in <module>
    import tomlib
ModuleNotFoundError: No module named 'tomlib'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dcramer/src/peated/.venv/bin/ptw", line 5, in <module>
    from pytest_watcher import run
  File "/home/dcramer/src/peated/.venv/lib/python3.11/site-packages/pytest_watcher/__init__.py", line 1, in <module>
    from .watcher import VERSION, run
  File "/home/dcramer/src/peated/.venv/lib/python3.11/site-packages/pytest_watcher/watcher.py", line 15, in <module>
    from .config import Config
  File "/home/dcramer/src/peated/.venv/lib/python3.11/site-packages/pytest_watcher/config.py", line 11, in <module>
    from pip._vendor import tomli as tomlib
ModuleNotFoundError: No module named 'pip'
```

This is in a pdm-based project. May operate diff than poetry and some other mechanisms.

**Comment by olzhasar** [comment]
Could you specify the exact version of Python you are using including the minor?

**Comment by dcramer** [comment]
3.11.1 - happy to test on a diff version if you want.  I'm a bit out of touch with how Python works these days so this error threw me a bit. Installing pip locally did resolve it tho.

**Comment by olzhasar** [comment]
Are you sure it wasn't an alpha version of Python 3.11? E.g. 3.11.0a2 or smth?
`tomlib` should be available in all releases of Python 3.11 unless it's a prerelease

**Comment by dcramer** [comment]
Let me circle back it might have uhh resolved itself trying to repro a third time..

(ignore my faliing test)

```
➜  ~/s/peated (feat/rewrite-api-in-python) ✔ python --version
Python 3.11.1
➜  ~/s/peated (feat/rewrite-api-in-python) ✔ pdm remove pip -d
Removing packages from dev dev-dependencies: pip
Changes are written to pyproject.toml.
b -d
🔒 Lock successful
Changes are written to pdm.lock.
Synchronizing working set with lock file: 0 to add, 0 to update, 1 to remove

  ✔ Remove peated 0.0.0 successful

🎉 All complete!

➜  ~/s/peated (feat/rewrite-api-in-python) ✔ pdm add tomlib -d
Adding packages to dev dev-dependencies: tomlib
See /tmp/pdm-lock-wa5i_9r8.log for detailed debug log.
[CandidateNotFound]: Unable to find candidates for tomlib. There may exist some issues with the package name or network condition.
Add '-v' to see the detailed traceback
➜  ~/s/peated (feat/rewrite-api-in-python) ✔ pdm add tomlib -d
➜  ~/s/peated (feat/rewrite-api-in-python) ✗ ptw .
➜  ~/s/peated (feat/rewrite-api-in-python) ✗ python
Python 3.11.1 (main, Jan  3 2023, 10:37:49) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tomlib
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'tomlib'
>>>
➜  ~/s/peated (feat/rewrite-api-in-python) ✗ ptw .
pytest-watcher version 0.3.3
Runner command: pytest --tb=short -p no:doctest
Waiting for file changes in /home/dcramer/src/peated
=================================================================================== test session starts ===================================================================================
platform linux -- Python 3.11.1, pytest-7.3.2, pluggy-1.2.0
rootdir: /home/dcramer/src/peated
configfile: pyproject.toml
testpaths: src
plugins: cov-4.1.0, mock-resources-2.7.0, Faker-18.11.1, anyio-3.7.0
collected 2 items

src/peated/api/endpoints/test_auth.py EE                                                                                                                                            [100%]

========================================================================================= ERRORS ==========================================================================================
______________________________________________________________________ ERROR at setup of test_auth_basic_valid_creds ______________________________________________________________________
.venv/lib/python3.11/site-packages/alembic/script/base.py:246: in _catch_revision_errors
    yield
.venv/lib/python3.11/site-packages/alembic/script/base.py:456: in _upgrade_revs
    for script in reversed(list(revs))
.venv/lib/python3.11/site-packages/alembic/script/revision.py:797: in iterate_revisions
    revisions, heads = fn(
.venv/lib/python3.11/site-packages/alembic/script/revision.py:1452: in _collect_upgrade_revisions
    current_revisions = self.get_revisions(lower)
.venv/lib/python3.11/site-packages/alembic/script/revision.py:525: in get_revisions
    return sum([self.get_revisions(id_elem) for id_elem in id_], ())
.venv/lib/python3.11/site-packages/alembic/script/revision.py:525: in <listcomp>
    return sum([self.get_revisions(id_elem) for id_elem in id_], ())
.venv/lib/python3.11/site-packages/alembic/script/revision.py:550: in get_revisions
    return tuple(
.venv/lib/python3.11/site-packages/alembic/script/revision.py:551: in <genexpr>
    self._revision_for_ident(rev_id, branch_label)
.venv/lib/python3.11/site-packages/alembic/script/revision.py:622: in _revision_for_ident
    raise ResolutionError(
E   alembic.script.revision.ResolutionError: No such revision or branch '7e21dc4c3d3b'
```

**Comment by olzhasar** [comment]
Looks weird. Let me know if you can reproduce this again

**Comment by dcramer** [comment]
```
➜  ~/s/peated (feat/rewrite-api-in-python) ✗ rm -rf .venv
➜  ~/s/peated (feat/rewrite-api-in-python) ✗ pdm install
The saved Python interpreter doesn't match the project's requirement. Trying to find another one.
python.use_venv is on, creating a virtualenv for this project...
Virtualenv is created successfully at /home/dcramer/src/peated/.venv
Synchronizing working set with lock file: 84 to add, 0 to update, 0 to remove

  ✔ Install setuptools 68.0.0 successful
  ✔ Install cachetools 5.3.1 successful
  ✔ Install certifi 2023.5.7 successful
  ✔ Install amqp 5.1.1 successful
  ✔ Install cfgv 3.3.1 successful
  ✔ Install billiard 4.1.0 successful
  ✔ Install anyio 3.7.0 successful
  ✔ Install click-didyoumean 0.3.0 successful
  ✔ Install click-plugins 1.1.1 successful
  ✔ Install cffi 1.15.1 successful
  ✔ Install charset-normalizer 3.1.0 successful
  ✔ Install click 8.1.3 successful
  ✔ Install click-repl 0.3.0 successful
  ✔ Install alembic 1.11.1 successful
  ✔ Install email-validator 2.0.0.post2 successful
  ✔ Install ecdsa 0.18.0 successful
  ✔ Install distlib 0.3.6 successful
  ✔ Install factory-boy 3.2.1 successful
  ✔ Install coverage 7.2.7 successful
  ✔ Install filelock 3.12.2 successful
  ✔ Install black 23.3.0 successful
  ✔ Install geoalchemy2 0.13.3 successful
  ✔ Install fastapi 0.97.0 successful
  ✔ Install celery 5.3.1 successful
  ✔ Install h11 0.14.0 successful
  ✔ Install dnspython 2.3.0 successful
  ✔ Install httptools 0.5.0 successful
  ✔ Install httpcore 0.17.2 successful
  ✔ Install iniconfig 2.0.0 successful
  ✔ Install google-auth 2.20.0 successful
  ✔ Install identify 2.5.24 successful
  ✔ Install idna 3.4 successful
  ✔ Install httpx 0.24.1 successful
  ✔ Install markupsafe 2.1.3 successful
  ✔ Install mypy-extensions 1.0.0 successful
  ✔ Install greenlet 2.0.2 successful
  ✔ Install nodeenv 1.8.0 successful
  ✔ Install mako 1.2.4 successful
  ✔ Install packaging 23.1 successful
  ✔ Install pathspec 0.11.1 successful
  ✔ Install platformdirs 3.7.0 successful
  ✔ Install pluggy 1.2.0 successful
  ✔ Install kombu 5.3.1 successful
  ✔ Install pyasn1 0.5.0 successful
  ✔ Install pre-commit 3.3.3 successful
  ✔ Install cryptography 41.0.1 successful
  ✔ Install passlib 1.7.4 successful
  ✔ Install pycparser 2.21 successful
  ✔ Install pytest-cov 4.1.0 successful
  ✔ Install prompt-toolkit 3.0.38 successful
  ✔ Install pytest-watcher 0.3.3 successful
  ✔ Install pytest-mock-resources 2.7.0 successful
  ✔ Install psycopg2-binary 2.9.6 successful
  ✔ Install pyasn1-modules 0.3.0 successful
  ✔ Install python-dotenv 1.0.0 successful
  ✔ Install python-dateutil 2.8.2 successful
  ✔ Install pytest 7.3.2 successful
  ✔ Install python-jose 3.3.0 successful
  ✔ Install rsa 4.9 successful
  ✔ Install requests 2.31.0 successful
  ✔ Install python-multipart 0.0.6 successful
  ✔ Install six 1.16.0 successful
  ✔ Install sniffio 1.3.0 successful
  ✔ Install pyyaml 6.0 successful
  ✔ Install pydantic 1.10.9 successful
  ✔ Install starlette 0.27.0 successful
  ✔ Install sentry-sdk 1.25.1 successful
  ✔ Install typing-extensions 4.6.3 successful
  ✔ Install tenacity 8.2.2 successful
  ✔ Install sqlalchemy-stubs 0.4 successful
  ✔ Install uvicorn 0.22.0 successful
  ✔ Install urllib3 1.26.16 successful
  ✔ Install vine 5.0.0 successful
  ✔ Install watchdog 3.0.0 successful
  ✔ Install faker 18.11.1 successful
  ✔ Install wcwidth 0.2.6 successful
  ✔ Install ruff 0.0.274 successful
  ✔ Install watchfiles 0.19.0 successful
  ✔ Install websockets 11.0.3 successful
  ✔ Install virtualenv 20.23.1 successful
  ✔ Install sqlalchemy 2.0.16 successful
  ✔ Install tzdata 2023.3 successful
  ✔ Install uvloop 0.17.0 successful
  ✔ Install mypy 1.4.0 successful
Installing the project as an editable package...
  ✔ Install peated 0.0.0 successful

🎉 All complete!

➜  ~/s/peated (feat/rewrite-api-in-python) ✗ python
➜  ~/s/peated (feat/rewrite-api-in-python) ✗ ptw .
Traceback (most recent call last):
  File "/home/dcramer/src/peated/.venv/lib/python3.11/site-packages/pytest_watcher/config.py", line 9, in <module>
    import tomlib
ModuleNotFoundError: No module named 'tomlib'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/dcramer/src/peated/.venv/bin/ptw", line 5, in <module>
    from pytest_watcher import run
  File "/home/dcramer/src/peated/.venv/lib/python3.11/site-packages/pytest_watcher/__init__.py", line 1, in <module>
    from .watcher import VERSION, run
  File "/home/dcramer/src/peated/.venv/lib/python3.11/site-packages/pytest_watcher/watcher.py", line 15, in <module>
    from .config import Config
  File "/home/dcramer/src/peated/.venv/lib/python3.11/site-packages/pytest_watcher/config.py", line 11, in <module>
    from pip._vendor import tomli as tomlib
ModuleNotFoundError: No module named 'pip'

➜  ~/s/peated (feat/rewrite-api-in-python) ✗ python -c 'import pip'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'pip'
```

Its possible this is an artifact of how pdm works (no pip in the default env, and tomlib only comes from pip). Not sure how the standard venv stuff works these days.

**Comment by olzhasar** [comment]
Could you check if you can import `tomli` in yout python shell?
```sh
python
import tomli
```

**Comment by dcramer** [comment]
Using pyenv if helpful. this is in and out of the virtualenv

```
➜  ~/s/peated (feat/rewrite-api-in-python) ✔ python
Python 3.11.1 (main, Jan  3 2023, 10:37:49) [GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tomli
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'tomli'
>>> import tomlib
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'tomlib'
>>>
➜  ~/s/peated (feat/rewrite-api-in-python) ✔ cd ..
direnv: unloading
➜  ~/src python -c 'import tomlib'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'tomlib'
➜  ~/src python -c 'import tomli'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'tomli'
➜  ~/src python --version
Python 3.11.1
```

**Comment by olzhasar** [comment]
I was able to represent this using `pyenv`. Will think of a possible solution now

**Comment by olzhasar** [comment]
Fixed in version 0.3.4

@dcramer Thank you for reporting this
